libraries/rush-lib/src/logic/operations/OperationExecutionManager.ts

```diff
@@ -261,8 +261,7 @@ export class OperationExecutionManager {
          */
         if (operation.status === UNASSIGNED_OPERATION) {
```

Comment:
This `if` check will now be unreachable code.
We do, however, need something to periodically ping `assignOperations` since the assignability of the operations queue can change suddenly; I forget if that exists already.

Reply:
Ah, that's what I was running into. Was seeing a deadlock ~50% of the time because an operation would finish but there would be no way to get an item to queue. I'll take a look at adding this tomorrow.

Reply:
Updated in [5d86108](https://github.com/microsoft/rushstack/pull/4821/commits/5d86108a1ca39a30a31a8917b9d6f8c6ef0ae131) and [ad619fe](https://github.com/microsoft/rushstack/pull/4821/commits/ad619fea15f21f368a5d80cce61f0f413ef76365)